dm/dm/master/scheduler/worker.go

```diff
@@ -141,74 +138,60 @@ func (w *Worker) ToBound(bound ha.SourceBound) error {
 	if w.stage == WorkerOffline {
 		return terror.ErrSchedulerWorkerInvalidTrans.Generate(w.BaseInfo(), WorkerOffline, WorkerBound)
 	}
-	if w.stage == WorkerRelay {
-		if w.relaySource != bound.Source {
-			return terror.ErrSchedulerBoundDiffWithStartedRelay.Generate(w.BaseInfo().Name, bound.Source, w.relaySource)
-		}
-	}
 
-	w.stage = WorkerBound
 	w.reportMetrics()
-	w.bound = bound
+	w.bounds[bound.Source] = bound
+	w.stage = WorkerBound
 	return nil
 }
 
+func (w *Worker) checkFree() {
+	if w.stage == WorkerBound && len(w.bounds) == 0 && len(w.relaySources) == 0 {
+		w.stage = WorkerFree
+	}
+}
+
 // Unbound changes worker's stage from Bound to Free or Relay.
-func (w *Worker) Unbound() error {
+func (w *Worker) Unbound(source string) error {
 	w.mu.Lock()
 	defer w.mu.Unlock()
-	if w.stage != WorkerBound {
+	if _, ok := w.bounds[source]; !ok {
 		// caller should not do this.
-		return terror.ErrSchedulerWorkerInvalidTrans.Generatef("can't unbound a worker that is not in bound stage.")
+		return terror.ErrSchedulerWorkerInvalidTrans.Generatef("can't unbound a source %s that is not bound with worker %s.", source, w.baseInfo.Name)
```

Comment:
it's time for us to start to use correct verb! "bind"/"unbind"/"be bound"

Reply:
correct verb✖️

correct tense☑️😂

Reply:
> correct verb✖️
> 
> correct tense☑️😂

I guess some developer treat bound as a verb, because there's "bounded" in code comments

Reply:
Feel confused the first time I saw it.